### PR TITLE
OSOE-74: Add UI test project

### LINKED ISSUE #25: Add UI testing extension method (OSOE-74)
Similar to Data Tables, add a UI testing extension method that checks what the [UI testing sample project's `ApplicationInsightsTrackingShouldBePresent` test does](https://github.com/Lombiq/UI-Testing-Toolbox/blob/dev/Lombiq.Tests.UI.Samples/Tests/BasicTests.cs#L125). Execute this from the OSOCE UI test project, and change the sample test to also do that, as an example of using external UI test methods.

Move the `EnableApplicationInsightsOfflineOperation` extension method from the Toolbox from this project, and use this from both test projects.

Publish the test project on NuGet and use it from the NuGet solution.

### PR DESCRIPTION
**Pull request by Piedone**
OSOE-74
Fixes #25

### CODE REVIEW

File: Lombiq.Hosting.Azure.ApplicationInsights.Tests.UI/Extensions/TestCaseUITestContextExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+using Shouldly;
+
+namespace Lombiq.Hosting.Azure.ApplicationInsights.Tests.UI.Extensions;
+
+public static class TestCaseUITestContextExtensions
+{
+    /// <summary>
+    /// Tests Application Insights tracking during offline operation. Use <see
+    /// cref="OrchardCoreUITestExecutorConfigurationExtensions.EnableApplicationInsightsOfflineOperation"/> to switch
+    /// the module into offline mode.
```
Comment (sarahelsaig) [inline Lombiq.Hosting.Azure.ApplicationInsights.Tests.UI/Extensions/TestCaseUITestContextExtensions.cs:14]: Shouldn't we verify this during runtime explicitly? For example add a `internal static IsOfflineOperationEnabled { get; private set; }` to `OrchardCoreUITestExecutorConfigurationExtensions`. Then `EnableApplicationInsightsOfflineOperation` sets it to `true`. This method checks and throws an exception with instructions if it's `false`. I see there is a message in the `ShouldBe`, but that's generic and doesn't tell about `EnableApplicationInsightsOfflineOperation`.

**Review by sarahelsaig — CHANGES_REQUESTED** [review]

File: Lombiq.Hosting.Azure.ApplicationInsights.Tests.UI/Extensions/TestCaseUITestContextExtensions.cs (hunk 1 above)
Comment (Piedone) [reply Lombiq.Hosting.Azure.ApplicationInsights.Tests.UI/Extensions/TestCaseUITestContextExtensions.cs:14]: Yes, good point.

File: Lombiq.Hosting.Azure.ApplicationInsights.Tests.UI/Extensions/TestCaseUITestContextExtensions.cs [hunk 2]
```diff
@@ -14,6 +15,15 @@ public static class TestCaseUITestContextExtensions
     /// </summary>
     public static void TestApplicationInsightsTrackingInOfflineOperation(this UITestContext context)
     {
+        if (!OrchardCoreUITestExecutorConfigurationExtensions.IsOfflineOperationEnabled)
+        {
+            throw new InvalidOperationException(
+                "Testing Application Insights tracking during offline operation only works if you switch the module " +
+                "into offline mode by calling " +
+                "OrchardCoreUITestExecutorConfigurationExtensions.EnableApplicationInsightsOfflineOperation before " +
```
Comment (sarahelsaig) [inline Lombiq.Hosting.Azure.ApplicationInsights.Tests.UI/Extensions/TestCaseUITestContextExtensions.cs]: Use `nameof`?

Comment (Piedone) [reply Lombiq.Hosting.Azure.ApplicationInsights.Tests.UI/Extensions/TestCaseUITestContextExtensions.cs]: Eh. Changed it.

**Review by sarahelsaig — APPROVED** [review]

**Pull request merged**